scripts/eval/eval.py

```diff
@@ -160,123 +159,75 @@ def evaluate_model(
         torch.cuda.synchronize()
     b = time.time()
 
-    log.info(f'Ran {model_cfg.model_name} eval in: {b-a} seconds')
+    log.info(f'Ran {model_name} eval in: {b-a} seconds')
     return (trainer, logger_keys, eval_gauntlet_callback, eval_gauntlet_df)
 
 
 def main(cfg: DictConfig) -> Tuple[List[Trainer], pd.DataFrame]:
+    cfgs: Tuple[Dict[str, Any], EvalConfig] = make_dataclass_and_log_config(
+        cfg, EvalConfig, EVAL_CONFIG_KEYS, icl_tasks_required=True)
+    logged_cfg, eval_config = cfgs
+
     # Run user provided code if specified
-    code_paths = pop_config(cfg,
-                            'code_paths',
-                            must_exist=False,
-                            default_value=[],
-                            convert=True)
-    for code_path in code_paths:
+    for code_path in (eval_config.code_paths or []):
         import_file(code_path)
 
-    om.resolve(cfg)
-
-    # Create copy of config for logging
-    logged_cfg: DictConfig = copy.deepcopy(cfg)
-
-    model_configs: ListConfig = pop_config(cfg, 'models', must_exist=True)
-    eval_gauntlet_config: Optional[Union[str, DictConfig]] = pop_config(
-        cfg, 'eval_gauntlet', must_exist=False, default_value=None)
-
-    fsdp_dict_cfg: Optional[DictConfig] = pop_config(cfg,
-                                                     'fsdp_config',
-                                                     must_exist=False,
-                                                     default_value=None)
-    fsdp_config: Optional[Dict] = om.to_container(
-        fsdp_dict_cfg,
-        resolve=True) if fsdp_dict_cfg is not None else None  # type: ignore
-    assert isinstance(fsdp_config, Dict) or fsdp_config is None
+    model_configs = to_list_container(eval_config.models)
```

Comment:
why do all these things have `to_container` calls, when you already called `to_container` on the main config?